### ISSUE TITLE
TokenQueue.unescape doesn't handle backslash correctly

### ISSUE THREAD

**Issue by kemezrp** [jhy/jsoup#581]
Lets say that I have escaped text representing `\\`. It was escaped in a way that each backslash was escaped with another backslash, so this text is now `\\\\` which needs to be written as string `"\\\\\\\\"` (because each of backslashes is also special character in string) 

Now when I try to unescape `"\\\\\\\\"` to `\\` with `TokenQueue.unescape("\\\\\\\\")` I am getting as result `\\\`. 

It happens because when backslash is placed in StringBuilder it is also placed in `last` character which allows it to become escaping symbol for next character. This causes String representing `\\\\` to be treated as text containing three pairs of `\\`.

We can solve this problem by rewriting this method into something like

```
public static String unescape(String in) {

    StringBuilder out = new StringBuilder();
    boolean isEscaped = false;

    for (char c : in.toCharArray()) {
        if (isEscaped) {        // handle escaping
            out.append(c);
            isEscaped = false;
        } else if (c == ESC) {  // don't add this `\` but remember to escape next character
            isEscaped = true;
        } else{                 // handle other characters normally
            out.append(c);
        }
    }
    return out.toString();
}
```

BTW: `last != 0` part in condition `if (last != 0 && last == ESC)`  is redundant since `last == ESC` can only be true if `last` is equal to `ESC` which automatically exclude possibility of `last` being `0` (and even if `ESC` will be `0` then this condition can't be true, because it would represent `x!=0 && x==0` which is always false, at least for primitive types).

**Comment by kemezrp** [comment]
Actually now that I think about it this report may be example of XY problem. I saw incorrect behaviour of this method when I tried to create selector for paragraph with `\` and `)`. When I used `select("p.containsOwn(\\))") it worked fine so I suspected that to represent `\` I would have to use `select("p.containsOwn(\\\\)")` but it can't returns empty Elements. I also tried using `select("p:matchesOwn(wanted \\\\)")` but it generates `java.util.regex.PatternSyntaxException: Unmatched closing ')`. Only thing which worked was `select("p:matchesOwn(wanted [\\\\])")`
